src/ocm/components/clusterConfiguration/staticIp/commonValidationSchemas.tsx

```diff
@@ -144,3 +145,23 @@ export const getIpAddressInSubnetValidationSchema = (
     },
   );
 };
+
+export const getIpIsNotNetworkOrBroadcastAddressSchema = (
+  protocolVersion: 'ipv4' | 'ipv6',
+  subnet: string,
+) => {
+  return Yup.string().test(
+    'is-not-network-or-broadcast',
+    `The IP address must not match the network or broadcast address`,
+    (value) => {
+      const subnetAddr = getAddressObject(subnet, protocolVersion);
+      if (subnetAddr) {
+        const subnetStart = subnetAddr?.startAddress().correctForm();
+        const subnetEnd = subnetAddr?.endAddress().correctForm();
+        if (value === subnetStart || value === subnetEnd) return false;
+        else return true;
```

Comment:
Can we change it to something like this? I don't see the need for an if/else statement if we are only returning a value based solely on its condition.

```suggestion
return !(value === subnetStart || value === subnetEnd);
```

Reply:
changed it!